Mathlib/Topology/Algebra/ClopenNhdofOne.lean

```diff
@@ -0,0 +1,36 @@
+/-
+Copyright (c) 2024 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan, Yi Song, Xuchun Li
+-/
+import Mathlib.GroupTheory.Index
+import Mathlib.Topology.Algebra.ClosedSubgroup
+import Mathlib.Topology.Algebra.OpenSubgroup
+/-!
+## The Open Normal Subgroup in a Clopen Neighborhood of One
+This file define `OpenNormalSubgroupSubClopenNhdsOfOne` , which strengthen the existing result of
+`OpenSubgroupSubClopenNhdsOfOne` into open normal subgroup.
+
+This file is split out because the need of more imports.
+-/
+
+namespace TopologicalGroup
+
+/-- The open normal subgroup contained in a clopen nhd of `1` in a compact topological group. -/
```

Comment:
This is not the unique such subgroup. Is it the maximal one? If so, please add that to the docstring. Can you also prove that?

Reply:
Sorry that for the current proof, I can't prove that, so what should the proof or the docstring be changed into?

Reply:
Yes, I think that would be better.

But what do you mean with "for the current proof, I can't prove that"?

What would need to change, to get a good characterization of this group?

Reply:
The proof used compactness to pick a finite subcover and then use it as index to intersect a family of open sets to get another open set, I don't really now how to change to make it have properties more than existence.

Reply:
In that case I suggest reflecting that in the statement, and proving a `Nonempty ...` instance.

Reply:
Sorry, what do you mean by "reflecting that in the statement"?

Reply:
Aahrg, sorry. I forgot about this. Just make it a `def` instead of an `instance`.

Reply:
It failed again with linter `DEF/LEMMA`, maybe it should be changed into a lemma